### PR TITLE
Fix getFood

### PR DESCRIPTION
**Pull request by GoogeTan**
I did it: https://forums.minecraftforge.net/topic/91558-1152-edit-minecraft-source-code-or-more-simply-hooks/?tab=comments
GetFood method now has an item stack

### CODE REVIEW

**Comment by ChampionAsh5357** [comment]
Already a pr #7543

File: patches/minecraft/net/minecraft/item/Item.java.patch [hunk 1]
```diff
@@ -121,6 +127,17 @@
...
++   @Deprecated
+   @Nullable
+   public Food func_219967_s() {
+      return this.field_219974_q;
+   }
++
++   @Nullable
++   public Food getFood(ItemStack stack) {
++      return this.field_219974_q;
++   }
```
Comment (pupnewfster) [inline patches/minecraft/net/minecraft/item/Item.java.patch:140]: a. New methods like this should be in IForgeItem
b. This is a breaking change as if any mod overrides `getFood` this then doesn't gracefully fallback to that override

File: patches/minecraft/net/minecraft/item/Item.java.patch [hunk 2]
```diff
@@ -121,6 +127,17 @@
     public boolean func_219970_i(ItemStack p_219970_1_) {
        return p_219970_1_.func_77973_b() == Items.field_222114_py;
     }
+@@ -306,4 +349,10 @@
++   @Deprecated
```
Comment (pupnewfster) [inline patches/minecraft/net/minecraft/item/Item.java.patch:131]: When we add deprecation in forge to vanilla methods we put a comment that says what to use instead.

**Review by pupnewfster — COMMENTED** [review]
I am going to close this PR given #7543 is already open, assigned and more complete, and does not have any merge conflicts (though I did notice from looking at this that a couple patch files I made didn't commit which I fixed now). But for future reference if you make further PRs I went ahead and reviewed this so you know how some things work/that would need to be changed.

Additionally some of your patch files look a bit weird with end of file line changing, and not as much surrounding code being added almost as if you edited them manually (which is a no go).

**Pull request closed without merging**